invisible_cities/cities/esmeralda_test.py

```diff
@@ -298,14 +235,108 @@ def test_esmeralda_exact_result_all_events(ICDATADIR, KrMC_hdst_filename, correc
+                         threshold_charge_paolina = 30   * units.pes       ,
+                         same_peak                = True                   ,
+                         apply_temp               = False                 ),
+                     paolina_params               = dict(
+                         vox_size                 = [15 * units.mm] * 3    ,
+                         strict_vox_size          = False                  ,
+                         energy_threshold         = 0 * units.keV         ,
+                         min_voxels               = 2                      ,
+                         blob_radius              = 21 * units.mm)        ))
+    cnt = esmeralda(**conf)
+
+    df_tracks = dio.load_dst(PATH_OUT, 'Tracking', 'Tracks')
+    df_phits  = dio.load_dst(PATH_OUT, 'CHITS'   , 'highTh')
+
+    for (event_num, ev_phits) in df_phits.groupby('event'):
+        assert  sum(df_tracks[df_tracks.event==event_num].numb_of_hits) == len(ev_phits)
+
+
+#test showing that all events that pass charge threshold are contained in hits output
+def test_esmeralda_all_hits_after_drop_voxels(data_hdst, esmeralda_tracks, correction_map_filename, config_tmpdir):
+    PATH_IN   = data_hdst
+    PATH_OUT  = os.path.join(config_tmpdir, "exact_tracks_esmeralda_drop_voxels.h5")
+    conf      = configure('dummy invisible_cities/config/esmeralda.conf'.split())
+    nevt_req  = all_events
+    th_p      = 30 * units.pes
+    conf.update(dict(files_in                     = PATH_IN                ,
+                     file_out                     = PATH_OUT               ,
+                     event_range                  = nevt_req               ,
+                     run_number                   = 6822                   ,
+                     cor_hits_params              = dict(
+                         map_fname                = correction_map_filename,
+                         threshold_charge_reco    = 10   * units.pes       ,
+                         threshold_charge_paolina = th_p                   ,
+                         same_peak                = True                   ,
+                         apply_temp               = False                 ),
+                     paolina_params               = dict(
+                         vox_size                 = [15 * units.mm] * 3    ,
+                         strict_vox_size          = False                  ,
+                         energy_threshold         = 20 * units.keV         ,
+                         min_voxels               = 2                      ,
+                         blob_radius              = 21 * units.mm)        ))
+    cnt = esmeralda(**conf)
+
+    df_tracks           =  dio.load_dst(PATH_OUT, 'Tracking', 'Tracks')
+    df_phits            =  dio.load_dst(PATH_OUT,   'CHITS' , 'highTh')
+    df_in_hits          =  dio.load_dst(PATH_IN ,    'RECO' , 'Events')
+
+    num_pass_th_in_hits = sum(df_in_hits.Q >= th_p)
```

Comment:
```suggestion
    num_pass_th_in_hits = np.nansum(df_in_hits.Q >= th_p)
```
I suggest this change to avoid a warning by numpy. I don't know if the hit charges can be nan

Reply:
if they are nan something went wrong so maybe is better to keep the sum and make the code complain?

Reply:
Indeed